### PR TITLE
PokemonDataGrid Component implementation

### PR DESCRIPTION
**Pull request by SebastianRosalesZ**

### CODE REVIEW

File: src/components/PokemonDataGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                        },
+                    }}
+                    pageSizeOptions={[20]}
+                    disableRowSelectionOnClick
+                    hideFooter
+                />
+            </Box>
+        </ThemeProvider>
+    );
+}
```
Comment (PabloTercerosJala) [inline src/components/PokemonDataGrid.tsx:96]: GENERAL COMMENTS:
Correct branch name misspell
Squash the commits into a single one

File: src/components/PokemonDataGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                }}
+            >
+                Show
+            </Button>
+
+        ),
+    },
+];
+
+interface PokemonDataGridProps {
```
Comment (PabloTercerosJala) [inline src/components/PokemonDataGrid.tsx]: Move interface to the top

File: src/components/PokemonDataGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+export default function PokemonDataGrid({ rows }: PokemonDataGridProps) {
+    const prefersDarkMode =
+        window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
+    const theme = React.useMemo(
+        () =>
+            createTheme({
+                palette: {
+                    mode: prefersDarkMode ? 'dark' : 'light',
```
Comment (PabloTercerosJala) [inline src/components/PokemonDataGrid.tsx]: The style does not match the sketch (the sketch style is white)

File: src/components/PokemonDataGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                },
+            }),
+        [prefersDarkMode]
+    );
+
+    return (
+        <ThemeProvider theme={theme}>
+            <Box sx={{ height: 400, width: '90%', maxWidth: '560px', margin: '0 auto' }}>
+                <DataGrid
+                    rows={rows}
```
Comment (PabloTercerosJala) [inline src/components/PokemonDataGrid.tsx:80]: The example rows are not fitting in the data grid

File: src/components/PokemonDataGrid.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+        ),
+    },
+];
+
+interface PokemonDataGridProps {
+    rows: { image: string; name: string; type: string }[];
+}
+
+export default function PokemonDataGrid({ rows }: PokemonDataGridProps) {
```
Comment (PabloTercerosJala) [inline src/components/PokemonDataGrid.tsx:51]: Sonar lint warning

File: package.json [hunk 6]
```diff
@@ -5,23 +5,24 @@
   "homepage": "https://pokejalasoft.github.io/PokeJalasoftProject/",
   "type": "module",
   "scripts": {
-    "predeploy" : "npm run build",
-    "deploy" : "gh-pages -d dist",
+    "predeploy": "npm run build",
```
Comment (PabloTercerosJala) [inline package.json:8]: Restore spaces

**Review by PabloTercerosJala — CHANGES_REQUESTED** [review]

**Review by PabloTercerosJala — COMMENTED** [review]
Do not upload package-json.lock

File: package.json (hunk 6 above)
Comment (SebastianRosalesZ) [reply package.json:8]: Actually not delete it there are top down

File: src/components/PokemonDataGrid.tsx (hunk 4 above)
Comment (SebastianRosalesZ) [reply src/components/PokemonDataGrid.tsx:80]: Elaborate more please or upload a picture

File: src/components/PokemonDataGrid.tsx (hunk 5 above)
Comment (SebastianRosalesZ) [reply src/components/PokemonDataGrid.tsx:51]: Elaborate more please

**Pull request closed without merging**